ibm/config.go

```diff
@@ -594,91 +594,368 @@ func (sess clientSession) IAMNamespaceAPI() (*ns.IbmCloudFunctionsNamespaceAPIV1
+	if sess.cisRLErr == nil {
+		cisRLOpt := &cisratelimitv1.ZoneRateLimitsV1Options{
+			URL:            sess.cisRLClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisRLClient.Service.Options.Authenticator,
+		}
+		cisRLClient, cisRLErr := cisratelimitv1.NewZoneRateLimitsV1(cisRLOpt)
+		if cisRLErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while cofiguring CIS Zone Rate Limit client: %s",
+				cisRLErr)
+		}
+		return cisRLClient, nil
+	}
 	return sess.cisRLClient, sess.cisRLErr
 }
 
 // CIS IP
 func (sess clientSession) CisIPClientSession() (*cisipv1.CisIpApiV1, error) {
+	if sess.cisIPErr == nil {
+		cisIPOpt := &cisipv1.CisIpApiV1Options{
+			URL:           sess.cisIPClient.Service.Options.URL,
+			Authenticator: sess.cisIPClient.Service.Options.Authenticator,
+		}
+		cisIPClient, cisIPErr := cisipv1.NewCisIpApiV1(cisIPOpt)
+		if cisIPErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS IP client: %s",
+				cisIPErr)
+		}
+		return cisIPClient, nil
+	}
 	return sess.cisIPClient, sess.cisIPErr
 }
 
 // CIS Page Rules
 func (sess clientSession) CisPageRuleClientSession() (*cispagerulev1.PageRuleApiV1, error) {
+	if sess.cisPageRuleErr == nil {
+		cisPageRuleOpt := &cispagerulev1.PageRuleApiV1Options{
+			URL:           sess.cisPageRuleClient.Service.Options.URL,
+			Crn:           core.StringPtr(""),
+			ZoneID:        core.StringPtr(""),
+			Authenticator: sess.cisPageRuleClient.Service.Options.Authenticator,
+		}
+		cisPageRuleClient, cisPageRuleErr := cispagerulev1.NewPageRuleApiV1(cisPageRuleOpt)
+		if cisPageRuleErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while cofiguring CIS Page Rule client: %s",
+				cisPageRuleErr)
+		}
+		return cisPageRuleClient, nil
+	}
 	return sess.cisPageRuleClient, sess.cisPageRuleErr
 }
 
-// cCIS Edge Function
+// CIS Edge Function
 func (sess clientSession) CisEdgeFunctionClientSession() (*cisedgefunctionv1.EdgeFunctionsApiV1, error) {
+	if sess.cisEdgeFunctionErr == nil {
+		cisEdgeFunctionOpt := &cisedgefunctionv1.EdgeFunctionsApiV1Options{
+			URL:            sess.cisEdgeFunctionClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisEdgeFunctionClient.Service.Options.Authenticator,
+		}
+		cisEdgeFunctionClient, cisEdgeFunctionErr :=
+			cisedgefunctionv1.NewEdgeFunctionsApiV1(cisEdgeFunctionOpt)
+		if cisEdgeFunctionErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Edge Function client: %s",
+				cisEdgeFunctionErr)
+		}
+		return cisEdgeFunctionClient, nil
+	}
 	return sess.cisEdgeFunctionClient, sess.cisEdgeFunctionErr
 }
 
 // CIS SSL certificate
 func (sess clientSession) CisSSLClientSession() (*cissslv1.SslCertificateApiV1, error) {
+	if sess.cisSSLErr == nil {
```

Comment:
Same here:

```go
if sess.cisSSLErr != nil {
	return sess.cisSSLClient, sess.cisSSLErr
}

cisSSLOpt := &cissslv1.SslCertificateApiV1Options{
	URL:            sess.cisSSLClient.Service.Options.URL,
	Crn:            core.StringPtr(""),
	ZoneIdentifier: core.StringPtr(""),
	Authenticator:  sess.cisSSLClient.Service.Options.Authenticator,
}

cisSSLClient, cisSSLErr := cissslv1.NewSslCertificateApiV1(cisSSLOpt)
if cisSSLErr != nil {
	return nil, fmt.Errorf(
		"Error occured while configuring CIS SSL certificate client: %s",
		cisSSLErr)
}

return cisSSLClient, nil

```

Reply:
I have upgraded sdk and regenerated cis sdk. Modified the code as per your suggession.